runtime/jsTest/src/kotlinx/serialization/DynamicSerializerTest.kt

```diff
@@ -0,0 +1,146 @@
+package kotlinx.serialization
+
+import kotlinx.serialization.internal.IntArraySerializer
+import kotlinx.serialization.internal.IntSerializer
+import kotlinx.serialization.json.Json
+import kotlin.test.*
+
+
+class DynamicSerializerTest {
+    @Serializable
+    data class Data(val a: Int)
+
+    @Serializable
+    open class DataWrapper(open val s: String, val d: Data?)
```

Comment:
Same question about `open`

Reply:
The test shows that the names are not mangled after serialization.